File: BigSemantics.GeneratedClassesCSharp/RichArtifactsTypeScope.cs
using System;
using Ecologylab.BigSemantics.Generated.Library.PrimitivesNS;
using Ecologylab.BigSemantics.MetadataNS.Builtins;
using Simpl.Serialization;

namespace Ecologylab.BigSemantics.Generated
{
    public class RichArtifactsTypeScope
    {
        public const string ScopeName = "rich_artifacts_scope";

        protected static readonly Type[] Translations =
            {
                typeof (RichArtifact<>),
                typeof (Clipping<>),
                typeof (ImageClipping),
                typeof (TextClipping),
                typeof (VideoClipping),
                typeof (MacheClipping),
                typeof (ImageSelfmade),
                typeof (TextSelfmade),
                typeof (WebVideo),
                typeof (HtmlText),
                typeof (Image),
                typeof (Video),
                typeof (Audio),
                typeof (FrameSelfmade),
                typeof (PresentationFrame),
                typeof (AggregateMedia),
                typeof (AggregateSelfmade)
            };

        public static SimplTypesScope Get()
        {
            return SimplTypesScope.Get(ScopeName, Translations);
        }
    }
}

File: BigSemanticsGeneratedClassesJava/src/ecologylab/bigsemantics/RichArtifactsTypeScope.java
/**
 * 
 */
package ecologylab.bigsemantics;

import ecologylab.bigsemantics.generated.library.primitives.VideoClipping;
import ecologylab.bigsemantics.generated.library.primitives.WebVideo;
import ecologylab.bigsemantics.metadata.builtins.Audio;
import ecologylab.bigsemantics.metadata.builtins.Clipping;
import ecologylab.bigsemantics.metadata.builtins.HtmlText;
import ecologylab.bigsemantics.metadata.builtins.Image;
import ecologylab.bigsemantics.metadata.builtins.ImageClipping;
import ecologylab.bigsemantics.metadata.builtins.ImageSelfmade;
import ecologylab.bigsemantics.metadata.builtins.RichArtifact;
import ecologylab.bigsemantics.metadata.builtins.TextClipping;
import ecologylab.bigsemantics.metadata.builtins.TextSelfmade;
import ecologylab.bigsemantics.metadata.builtins.Video;
import ecologylab.serialization.SimplTypesScope;

/**
 * @author andrew
 *
 */
public class RichArtifactsTypeScope {

	public static final String SCOPE_NAME = "rich_artifacts_scope";

    protected static final Class[] TRANSLATIONS =
        {
            RichArtifact.class,
            Clipping.class,
            ImageClipping.class,
            TextClipping.class,
            VideoClipping.class,
            ImageSelfmade.class,
            TextSelfmade.class,
        };

    public static SimplTypesScope get()
    {
        return SimplTypesScope.get(SCOPE_NAME, RichMediaTypeScope.get(), TRANSLATIONS);
    }
}
